### PR TITLE
Create Kadane's Algorithm.md

### LINKED ISSUE #919: [General] Add Kadane’s Algorithm for Maximum Subarray Sum
### Issue Description
Kadane’s algorithm is a popular algorithm used to find the **maximum sum of a contiguous subarray** in an array with a time complexity of **O(n)**. It efficiently handles both positive and negative numbers, making it an essential part of algorithmic problem-solving.
This algorithm can be a valuable addition to the project’s library under the **"Array Algorithms"** section, improving resources for learners and contributors.

**Would you like to work on this issue?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by Siddhart2004**
### Description
This Pull Request implements Kadane's Algorithm, which efficiently finds the maximum sum of a contiguous subarray within a one-dimensional numeric array.

Fixes #919

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Siddhart2004** [comment]
@ajay-dhangar, I have completed the pr, Kindly review it and provide feedback also gssoc and hacktoberfest labels.

File: arrays/Kadane's Algorithm.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+id: kadanes-algorithm
+title: Kadane's Algorithm
```
Comment (ajay-dhangar) [inline arrays/Kadane's Algorithm.md]: replace `title: Kadane's Algorithm` with `title: "Kadane's Algorithm"`

File: arrays/Kadane's Algorithm.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+sidebar_label: Introduction to Kadane's Algorithm
```
Comment (ajay-dhangar) [inline arrays/Kadane's Algorithm.md]: replace `sidebar_label: Introduction to Kadane's Algorithm` with `sidebar_label: "Introduction to Kadane's Algorithm"`

File: arrays/Kadane's Algorithm.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+sidebar_position: 2
+description: Kadane's Algorithm efficiently finds the maximum sum of a contiguous subarray. It is a fundamental algorithm for solving problems involving subarray sums.
```
Comment (ajay-dhangar) [inline arrays/Kadane's Algorithm.md]: replace `description: Kadane's Algorithm efficiently finds the maximum sum of a contiguous subarray. It is a fundamental algorithm for solving problems involving subarray sums.` with `description: "Kadane's Algorithm efficiently finds the maximum sum of a contiguous subarray. It is a fundamental algorithm for solving problems involving subarray sums."`

**Review by ajay-dhangar — COMMENTED** [review]
update your code based on suggestions

**Comment by Siddhart2004** [comment]
@ajay-dhangar Made the changes according to your suggestions, also kindly add hactoberfest accepted label.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**